### PR TITLE
Migrate adaboost test from boost to catch2

### PR DESCRIPTION
**Pull request by jeffin143**
Refer : #2523

### CODE REVIEW

File: src/mlpack/tests/main_tests/adaboost_test.cpp [hunk 1]
```diff
@@ -41,25 +41,24 @@ struct AdaBoostTestFixture
...
 
-BOOST_FIXTURE_TEST_SUITE(AdaBoostMainTest, AdaBoostTestFixture);
-
 /**
  * Check that number of output labels and number of input
  * points are equal.
  */
-BOOST_AUTO_TEST_CASE(AdaBoostOutputDimensionTest)
+TEST_CASE_METHOD(AdaBoostTestFixture, "AdaBoostOutputDimensionTest",
+                 "AdaBoostMainTest")
```
Comment (rcurtin) [inline src/mlpack/tests/main_tests/adaboost_test.cpp:49]:
```suggestion
                 "[AdaBoostMainTest][BindingTests]")
```
We should probably do this for the other tests in this file too; what do you think?

**Comment by zoq** [comment]
Just resolved the merge conflict, let's see if the build comes back green here as well.

**Comment by zoq** [comment]
Ignored PVS - 521 over all files now - https://github.com/mlpack/jenkins-conf/commit/649282d0afcbfe9cd2abca4a0a5ed921f98211f6

**Comment by jeffin143** [comment]
@zoq , could hit the build button again, boost was not installed may be some network issue

**Comment by rcurtin** [comment]
Looks like the Linux builds failed to download Boost or something.  Let me close and reopen the PR to re-trigger the builds...

**Pull request merged**